httpclient5-cache/src/main/java/org/apache/hc/client5/http/impl/cache/CachingExec.java

```diff
@@ -507,29 +522,26 @@ ClassicHttpResponse negotiateResponseFromVariants(
             }
 
             final String resultEtag = resultEtagHeader.getValue();
-            final Variant matchingVariant = variants.get(resultEtag);
-            if (matchingVariant == null) {
+            final CacheHit match = variantMap.get(resultEtag);
+            if (match == null) {
                 LOG.debug("304 response did not contain ETag matching one sent in If-None-Match");
                 return callBackend(target, request, scope, chain);
             }
 
-            final HttpCacheEntry variantEntry = matchingVariant.getEntry();
-
-            if (revalidationResponseIsTooOld(backendResponse, variantEntry)
+            if (revalidationResponseIsTooOld(backendResponse, match.entry)
                     && (request.getEntity() == null || request.getEntity().isRepeatable())) {
                 final ClassicHttpRequest unconditional = conditionalRequestBuilder.buildUnconditionalRequest(request);
                 return callBackend(target, unconditional, scope, chain);
             }
 
             recordCacheUpdate(scope.clientContext);
 
-            final HttpCacheEntry responseEntry = responseCache.updateVariantEntry(
-                    target, conditionalRequest, backendResponse, variantEntry, requestDate, responseDate);
-            if (shouldSendNotModifiedResponse(request, responseEntry)) {
-                return convert(responseGenerator.generateNotModifiedResponse(responseEntry), scope);
+            final CacheHit hit = responseCache.update(match, backendResponse, requestDate, responseDate);
+            if (shouldSendNotModifiedResponse(request, hit.entry)) {
+                return convert(responseGenerator.generateNotModifiedResponse(hit.entry), scope);
             }
-            final SimpleHttpResponse response = responseGenerator.generateResponse(request, responseEntry);
-            responseCache.reuseVariantEntryFor(target, request, backendResponse, responseEntry, requestDate, responseDate);
+            final SimpleHttpResponse response = responseGenerator.generateResponse(request, hit.entry);
+            responseCache.storeReusing(hit, target, request, backendResponse, requestDate, responseDate);
```

Comment:
Shouldn't be the `entry` instead of the `hit`?

Reply:
@arturobernalg Done.